visualize_classification_groups.py

```diff
@@ -0,0 +1,206 @@
+		num_imgs = cartesian_coords.shape[0] if len(inds) >= cartesian_coords.shape[0] else len(inds) 
+		selected_inds = np.random.choice(inds, num_imgs, replace=False)				# choose a subset of max 30 elements from this set
+		print(f'Number of images in {slidename}: {len(inds)}')
+		blank_slide_layout = ppt.slide_layouts[6]
+		slide = ppt.slides.add_slide(blank_slide_layout)
+		txBox = slide.shapes.add_textbox(Inches(ppt.slide_height.inches/2), Inches(0.2), Inches(2), Inches(1))	# add slide title
+		txBox.text_frame.text = slidename
+		for j, ind in tqdm(enumerate(selected_inds)):
+			coord = cartesian_coords[j]
+			imgnew, label, imgold = dset[ind]
+			with tables.open_file(dset.fname, 'r') as f:	# we need to get the image_filename manually.
+				img_filename = f.root.filenames[ind]
+			plt.imshow(np.moveaxis(imgnew.numpy(), 0, -1))
+			plt.title(img_filename.decode("utf-8").split('/')[-1])
+			plt.tick_params(color='b',bottom=False, left=False)
+			plt.xticks([])
+			plt.yticks([])
+			with BytesIO() as img_buf:
+				plt.savefig(img_buf, format='png', bbox_inches='tight')
+				plt.close()
+				im = Image.open(img_buf)
+				addimagetoslide(slide, img_buf, top=Inches(coord[0]*1.2+1),left=Inches(coord[1]*1.2 + 0.25), width=Inches(im.height/im.width), height=Inches(1))
+				im.close()
+	ppt.save(pptx_save_path)
+
+if __name__ == "__main__":
+	parser = argparse.ArgumentParser()
+	parser.add_argument('--h5path', type=str, help='The path to an h5 file that contains "patch", "ground_truth_label", and "fname" datasets.')	
+	parser.add_argument('--model_checkpoint', '-c', type=str, default=None, help='The path to a model checkpoint for the torch.load() method.')
+	parser.add_argument('--ppt_save', type=str, help='The full path to save the generated powerpoint.')
+
+	parser.add_argument('--patch_size', type=int, default=224, help='The width of a square patch.')
+	parser.add_argument('--gpuid', type=int, default=0, help='The device id.')
+	parser.add_argument('--str_criteria', '-s', default=["TRUE NEGATIVES", "FALSE NEGATIVES", "FALSE POSITIVES", "TRUE POSITIVES"], type=str, nargs="+")
+	parser.add_argument('--criteria', default=['00', '10', '01', '11'], type=str, nargs="+", help="Each item is a pair of ground-truth,prediction labels.")
+	args = parser.parse_args()
+
+	##### DENSENET PARAMS #####
+	num_classes=3    #number of classes in the data mask that we'll aim to predict
+	in_channels= 3  #input channel of the data, RGB = 3
+
+	growth_rate=32 
+	block_config=(2, 2, 2, 2)
+	num_init_features=64
+	bn_size=4
+	drop_rate=0
+
+	##### COMPOSE IMAGE TRANSFORM OBJECT #####
+	img_transform = Compose([	# we choose not to apply any image transforms.
+		# VerticalFlip(p=.5),
+		# HorizontalFlip(p=.5),
+		# HueSaturationValue(hue_shift_limit=(-25,0),sat_shift_limit=0,val_shift_limit=0,p=1),
+		# Rotate(p=1, border_mode=cv2.BORDER_CONSTANT,value=0),
+		RandomSizedCrop((args.patch_size,args.patch_size), args.patch_size,args.patch_size),
```

Comment:
seems like a strange choice, isn't the randomsized crop going to randomly resize and crop? did you look at the documentation? the range seems to be pretty large, from .08 to 1?

https://albumentations.ai/docs/api_reference/augmentations/crops/transforms/#albumentations.augmentations.crops.transforms.RandomResizedCrop

perhaps you want something more like CenterCrop or with padding?